Include readme on target package

thekiwi opened the pull request:
This tool is awesome, but I noticed it wasn't bringing the readme documents into GitHub packages. 

This change copies the readme from srcPackument to the target manifest, if one didn't exist already.

I thought someone else might find this useful :)

joebowbeer commented on src/index.ts:
```diff
@@ -34,6 +34,9 @@ export async function sync(name: string, from: Record<string, string>, to: Recor
     console.log('Reading %s from %s', spec, from.registry)
 
     const manifest = srcPackument.versions[version]
+    if (!manifest.readme) {
```
Would it be better to assign `latestReadme` like `latestRepository`above?
Then the fix-up can be performed by `prepareManifest`, where it is easier to unit test.

thekiwi replied:
Thanks for the comments! 

> AFAICT, this change is copying the readme from the latest version into every version where it is missing, which seems problematic.

This is a good point. Unfortunately, it seems that only the most recent readme is available. This is actually mentioned in the link you referenced:

```
readme: the first 64K of the README data for the most recently published version of the package
```

As such, I will amend the PR to only attach the readme to the latest version.

> Can you provide more details about what this change is fixing?

Sure! As mentioned, when I test-ran this to migrate a package from NPM to GitHub Packages, the target package in GitHub was missing a readme. While the readme already exists in the repo, there will be times when the repo has changes that haven't been published, so it's preferable to capture the readme at the time of publishing. 

> If needed, I would prefer to perform the fix-up in prepareManifest, where it is easier to unit test.
> If transferring readme then also transfer readmeFilename?

Both make sense. I'll make those changes and push a new commit.

joebowbeer reviewed (commented):
Thanks. See questions in index.

thekiwi commented:
I've committed my changes based on your feedback. There are a few more changes than I had hoped but I've updated and added new tests for the `prepareManifest` function.

joebowbeer reviewed (approved):
LGTM - publishing as 1.3.0

joebowbeer commented:
@thekiwi By the way, I see a related bug report in libnpmpublish:

https://github.com/npm/libnpmpublish/issues/12

I wonder if your changes were in essence working around a libnpmpublish issue?
